pkg/store/bucket.go

```diff
@@ -1284,6 +1300,15 @@ func newBucketBlock(
 		enablePostingsCompression: enablePostingsCompression,
 	}
 
+	// Translate the block's labels and inject the block ID as a label
+	// to allow to match blocks also by ID.
+	b.labels = labels.FromMap(meta.Thanos.Labels)
+	b.labels = append(b.labels, labels.Label{
```

Comment:
I hope this is not advertised? (: 

Also that's funny, because technically you could now compose query to filter blocks as well, no?

Reply:
(because we match normal requests against block external labels.)  I think that's not what we want...

Reply:
> I hope this is not advertised? (:

It's not.

> Also that's funny, because technically you could now compose query to filter blocks as well, no?

I don't think so. We only apply block matchers in the request hints (if any). The block's meta external labels are never manipulated, but a copy is done (via `labels.FromMap()`).